versioned_hdf5/api.py

```diff
@@ -67,6 +67,26 @@ def __init__(self, f):
         self.f = f
         if '_version_data' not in f:
             initialize(f)
+        else:
+            # This is not a new file; check data version identifier for compatibility
+            if self.data_version_identifier != DATA_VERSION:
```

Comment:
Should probably be a < check here. Actually there should maybe be some sort of error in the VersionedHDF5File constructor if the data version is greater than expected.

Reply:
Good idea, I've changed this to raise an exception if a user loads a file written by a future version as well.